test/e2e/deployment.go

```diff
@@ -389,8 +389,8 @@ func testRecreateDeployment(f *Framework) {
 	newRS, err := deploymentutil.GetNewReplicaSet(*deployment, c)
 	Expect(err).NotTo(HaveOccurred())
 	Expect(newRS).NotTo(Equal(nil))
-	Expect(events.Items[0].Message).Should(Equal(fmt.Sprintf("Scaled down replica set %s to 0", rsName)))
-	Expect(events.Items[1].Message).Should(Equal(fmt.Sprintf("Scaled up replica set %s to 3", newRS.Name)))
+	Expect(events.Items[0].Message).Should(Equal(fmt.Sprintf("Scaled up replica set %s to 3", newRS.Name)))
+	Expect(events.Items[1].Message).Should(Equal(fmt.Sprintf("Scaled down replica set %s to 0", rsName)))
```

Comment:
Scale up event happens before scale down when Recreate because we set rs.Spec.Replicas = deployment.Spec.Replicas when creating the new RS. 

However, in types.go, it says: 
```go
     // Kill all existing pods before creating new ones.                                                                                                   
     RecreateDeploymentStrategyType DeploymentStrategyType = "Recreate"
```

Should we set it to `0` instead when Recreate?